src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -0,0 +1,1305 @@
+    _output << std::left
+        << std::setw(col1) << "CU Name"
+        << " " << std::setw(col2) << "AXI Portname"
+        << "  " << std::setw(16) << "Overall Status"
+        << "  " << std::setw(16) << "Snapshot"
+        << "  " << std::setw(16) << "Current"
+        << std::endl;
+    for (size_t i = 0; i < spcResults.NumSlots; ++i) {
+      _output << std::left
+                << std::setw(col1) << cuNames[AXI_STREAM_PROTOCOL_CHECKER][i] 
+                << " " << std::setw(col2) << portNames[AXI_STREAM_PROTOCOL_CHECKER][i]
+                << "  " << std::setw(16) << std::hex << spcResults.PCAsserted[i]
+                << "  " << std::setw(16) << std::hex << spcResults.SnapshotPC[i]
+                << "  " << std::setw(16) << std::hex << spcResults.CurrentPC[i]
+                << std::dec << std::endl;
+    }
+    // Restore formatting
+    _output.copyfmt(saveFormat);
+  }
+
+  
+}
+
+
+void 
+DebugIpStatusCollector::populateOverview(boost::property_tree::ptree &_pt)
+{
+  if(!map || !map->m_count) {
+    std::cout << " INFO: Debug IP Data not populated." << std::endl;
+    return;
+  }
+  _pt.put("total_debug_ip_num", map->m_count);
+
+  for(uint64_t i = 0; i < map->m_count; i++) {
+    if (map->m_debug_ip_data[i].m_type > maxDebugIpType) {
+      std::cout << "Found invalid IP in debug ip layout with type "
+              << map->m_debug_ip_data[i].m_type << std::endl;
+      return;
+    }
+    ++debugIpNum[map->m_debug_ip_data[i].m_type];
+  }
+
+  for(uint8_t i = 0; i < maxDebugIpType; i++) {
+    if(0 == debugIpNum[i]) {
+       continue;
+    }
+    _pt.put(debugIpNames[i], debugIpNum[i]);
+  }
+}
+
+void 
+DebugIpStatusCollector::populateAllResults(boost::property_tree::ptree &_pt)
+{
+  populateAIMResults(_pt);
+  populateAMResults(_pt);
+  populateASMResults(_pt);
+  populateLAPCResults(_pt);
+  populateSPCResults(_pt);
+}
+
+
+void 
+DebugIpStatusCollector::populateAIMResults(boost::property_tree::ptree &_pt)
+{
+  if(0 == aimResults.NumSlots) {
+    return;
+  }
+
+  boost::property_tree::ptree aim_pt;
+
+  for(size_t i = 0; i < aimResults.NumSlots; ++i) {
+    boost::property_tree::ptree entry;
+    std::string entryStr = "AIM " + std::to_string(i);
+
+    entry.put("Region or CU", cuNames[AXI_MM_MONITOR][i]);
+    entry.put("Type or Port", portNames[AXI_MM_MONITOR][i]);
+    entry.put("Write kBytes", static_cast<double>(aimResults.WriteBytes[i])/1000.0);
+    entry.put("Write Trans",  aimResults.WriteTranx[i]);
+    entry.put("Read kBytes",  static_cast<double>(aimResults.ReadBytes[i])/1000.0);
+    entry.put("Read Tranx",   aimResults.ReadTranx[i]);
+    entry.put("Outstanding Cnt", aimResults.OutStandCnts[i]);
+    entry.put("Last Wr Addr", aimResults.LastWriteAddr[i]);
+    entry.put("Last Wr Data", aimResults.LastWriteData[i]);
+    entry.put("Last Rd Addr", aimResults.LastReadAddr[i]);
+    entry.put("Last Rd Data", aimResults.LastReadData[i]);
+
+    aim_pt.add_child(entryStr.c_str(), entry);
```

Comment:
Q: Can the entryString contain either a period '.' or forward slash '/'?
Not asking to change this value.  Asking for we need to determine a hierarchy search character that isn't being used by a key.

Reply:
No, the entryStr here willnot have any period '.' or forward slash '/'.